editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -7847,6 +7847,7 @@ void Node3DEditor::_sun_environ_settings_pressed() {
 	sun_environ_popup->set_position(pos - Vector2(sun_environ_popup->get_contents_minimum_size().width / 2, 0));
 	sun_environ_popup->reset_size();
 	sun_environ_popup->popup();
+	sun_environ_popup->grab_focus();
```

Comment:
Add a comment to explain why this is needed, as it's not obvious at a glance:

```suggestion
	// Grabbing the focus is required for Shift modifier checking to be functional
	// (when the Add sun/environment buttons are pressed).
	sun_environ_popup->grab_focus();
```